### ISSUE TITLE
Default fake behavior for ToString() doesn't produce the same type format as Object.ToString()

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#994]
The default `ToString()` method for a non-fake object returns `GetType().ToString()`, but for a fake it returns `"Faked" <faked type>.FullName`. For non-generic types, `ToString()` and `FullName` return the same thing, but for generic types `FullName` includes the fully qualified names of type arguments, i.e.

```
Console.WriteLine(typeof(List<string>).ToString()); // System.Collections.Generic.List`1[System.String]
Console.WriteLine(typeof(List<string>).FullName); // System.Collections.Generic.List`1[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]
```

Fakes use the second form, which is probably too verbose for most use cases, and is inconsistent with the normal `Object.ToString()`. We should probably consider using `Type.ToString()` instead of `Type.FullName` in the default fake behavior for `ToString()`.

**Comment by blairconrad** [comment]
I like it! In my opinion, this could be **ready** and **P3**, but I'm not that fussy about the priority.

**Comment by thomaslevesque** [comment]
My only worry is that it's arguably a breaking change, which means it should be in the next major release...

**Comment by blairconrad** [comment]
Ah. I see. I would not worry. We've changed messages before, I'm sure. And any change _can_ be a breaking change. I think relying on the format of error/exception messages is a little too much.

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 3.1.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/3.1.0).
